datafusion/src/sql/planner.rs

```diff
@@ -1062,8 +1062,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 }
 
                 let field = schema.field(field_index - 1);
-                let col_ident = SQLExpr::Identifier(Ident::new(field.qualified_name()));
-                self.sql_expr_to_logical_expr(&col_ident, schema)?
+                Expr::Column(field.qualified_column())
```

Comment:
This change avoids trying to reparse the string as a column name, and instead just creates the `Column` reference directly

Reply:
👍, FYI @hntd187

Reply:
This is much better than what I had :-)

Reply:
We are all learning together!